cpp/src/arrow/compute/exec/hash_join_graphs.py

```diff
@@ -0,0 +1,167 @@
+#!/bin/env python3
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+'''
+This script takes a JSON file from a google benchmark run that measures rows/sec and generates graphs
+for each benchmark.
+'''
+
+import math
+import sys
+import json
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+sns.set()
+
+class Test:
+    def __init__(self):
+        self.times = []
+        self.args = {}
+
+    def get_argnames_by_cardinality_increasing(self):
+        key_cardinality = lambda x: len(set(self.args[x]))
+        def key_strings_first(x):
+            try:
+                as_float = float(self.args[x][0])
+                return True
+            except:
+                return False
+        by_cardinality = sorted(self.args.keys(), key=key_cardinality)
+        strings_first = sorted(by_cardinality, key=key_strings_first)
+        return strings_first
+
+def organize_tests(filename):
+    tests = {}
+    with open(filename) as f:
+        df = json.load(f)
+        for idx, row in enumerate(df['benchmarks']):
+            test_name = row['name']
+            test_name_split = test_name.split('/')
+            if test_name_split[-1] == 'process_time':
+                test_name_split = test_name_split[:-1]
+                
+            base_name = test_name_split[0]
+            args = test_name_split[1:]
+            if base_name not in tests.keys():
+                tests[base_name] = Test()
+
+            tests[base_name].times.append(row['rows/sec'])
+            
+            if len(args) > 3:
+                raise('Test can have at most 3 parameters! Found', len(args), 'in test', test_name)
+            
+            nonnamed_args = [x for x in args if ':' not in x]
+            if len(nonnamed_args) > 1:
+                raise('Test name must have only one non-named parameter! Found', len(nonnamed_args), 'in test', test_name)
+
+            for arg in args:
+                arg_name = ''
+                arg_value = arg.strip('\"')
+                if ':' in arg:
+                    arg_split = arg.split(':')
+                    arg_name = arg_split[0]
+                    arg_value = arg_split[1].strip('\"')
```

Comment:
Since these are string values the order gets messed up since the UTF8 plot has unique X axis values.  On the other hand, if you read them in numerically, you'll need to configure the X axis to be logarithmic or else you'll get washed out by the higher values.

![badxaxis](https://user-images.githubusercontent.com/1696093/148860958-00adb09b-312b-465c-a3ea-333bd8178420.png)
![goodxaxis](https://user-images.githubusercontent.com/1696093/148860960-b87d4998-ddd7-4b95-9456-582f189b904b.png)